src/import/pull-tar.c

```diff
@@ -315,11 +315,13 @@ static void tar_pull_job_on_finished(PullJob *j) {
         if (!tar_pull_is_done(i))
                 return;
 
-        if (i->checksum_job->style == VERIFICATION_PER_DIRECTORY && i->signature_job->error != 0) {
-                log_error_errno(j->error, "Failed to retrieve signature file, cannot verify. (Try --verify=no?)");
+        if (i->signature_job) {
+                if (i->checksum_job->style == VERIFICATION_PER_DIRECTORY && i->signature_job->error != 0) {
```

Comment:
ok, but don't you also have to check whether checksum_job is != NULL too?

Also I think it would be nicer to just add this to the if check, but only place one if check here. i.e. instead of:

```c
if (i->signature_job) {
        if (i->checksum_job->… && i->signature_job…) {
                …
```

do:

```c
…
if (i->signature_job && i->checksum_job->… && i->signature_job…) {
        …
```

That saves one level of indentation..